backend/main.py

```diff
@@ -19,10 +37,33 @@ async def counties():
 
 
 @app.get("/counties/{county_id}")
-async def get_county_data(county_id: str):
+async def get_county_data(county_id: str) -> CountyResponse:
+    response = CountyResponse(message="Returning county data")
     if county_id in county_list:
         county: str = county_list[county_id]
-        county_data: pd.DataFrame = getData(county)
-        return {"data": county_data}
+
+        county_cache_exists = r.exists(county)
+        if (county_cache_exists == 1 ): # 1 = exists, 0 = does not exist
```

Comment:
Might be a little less readable, but do you want to do something like

`if r.exists(county): `